### DISCUSSION TITLE
Lineamientos para FORMULAR PREGUNTAS EFICIENTEMENTE

### DISCUSSION THREAD

**Discussion by Felipe-RA** [rramosp/ai4eng.v1#145] [00 - General]
Saludos estudiantes

Les adjuntamos estos lineamientos para agilizar el proceso de realizar preguntas y facilitar ayudarles lo más pronto posible:

*Ya sea de un ejercicio de laboratorios, de las notas, o de algo que se encontraron por ahí y no entienden:*

## ¡El formato general de una buena pregunta!

Incluyan suficiente información relacionada con su problema/duda: Por lo general, una buena pregunta tiene la siguiente información: 

- > Estaba Intentando hacer **X** pero obtuve algo inesperado
- > Me sale este error (adjunten el error):
- > Intenté solucionar el problema haciendo **Y** pero... / Después de buscar no encuentro cómo podría solucionar **X**
- > Este es mi código, creo que el problema puede estar en... (referencien la línea de código)

Todas estas son expresiones comunes que pueden utilizar para formular mejor sus preguntas, por supuesto no todas las preguntas se acomodan a este tipo de formato, pero sigan el principio general de **ser lo más informativos posibles**.

## Por favor utilicen bloques de código

Los bloques de código son siempre preferibles a capturas de pantalla, pueden usarlos para los mensajes de error también.

Los bloques de código facilitan la lectura del código que pongan en los post, también facilita poder copiarlo en caso de que queramos probar algo. Todo esto se traduce en que les podemos responder más rápido.

Les invitamos a que para facilidad de todos hagan uso de estos, esta es la sintaxis para lograrlo, usando \```  \```:

**EJEMPLO DE COMO PONERLO:**
\```
def create_deck(n_heaps, cards_per_heap, shuffle=False):
      n_cards = n_heaps * cards_per_heap 
      chars = [chr(i) for i in np.arange(26)+65]
      names = [i+j for i,j in itertools.product(chars, chars)]    
      assert n_cards < len(names), "cannot have more than %d cards"%len(name)
      c = np.r_[names[:n_cards]]
      if shuffle:
          c = np.random.permutation(c)
      return c    
\``` 

**RESULTADO DE LO QUE SE VÉ:**
```
def create_deck(n_heaps, cards_per_heap, shuffle=False):
    n_cards = n_heaps * cards_per_heap
    
    chars = [chr(i) for i in np.arange(26)+65]
    names = [i+j for i,j in itertools.product(chars, chars)]    

    assert n_cards < len(names), "cannot have more than %d cards"%len(name)
    
    c = np.r_[names[:n_cards]]
    if shuffle:
        c = np.random.permutation(c)
    return c    
``` 

Como pueden observar, el resultado es mucho más legible y conserva el formato original.

**Comment by soyudea** [comment]
[MOVED BY FELIPE TO DISCUSSION #217 ]
@soyudea

**Reply by Felipe-RA** [reply]
Saludos soyudea

Primero, decirte que por favor la próxima vez abras una nueva discusión.

Abriré una por tí y seguimos allí. La tienes aquí: #217
